src/app/ReadClient.cpp

```diff
@@ -1322,7 +1340,13 @@ void ReadClient::HandleDeviceConnectionFailure(void * context, const Operational
     _this->mMinimalResubscribeDelay = System::Clock::kZero;
 #endif // CHIP_CONFIG_ENABLE_BUSY_HANDLING_FOR_OPERATIONAL_SESSION_SETUP
 
-    _this->Close(failureInfo.error);
+    if (_this->IsPeerLIT() && err == CHIP_ERROR_TIMEOUT && _this->IsIdle())
```

Comment:
I don't get this state check.  If we are in this code, how could we be in a state other than "Idle"?

Reply:
move back to previous logic, no need for this change. thanks,